### PR TITLE
Support overall survival data from gdc

### PR DESCRIPTION
**Pull request by xzhou82**
## Description

at gdc dictionary, Overall Survival is added as the first root term. some improvements to querySamples_gdcapi()

image

adding it to oncomatrix and gene exp clustering seems to work

it doesn't work for [mds3 subtk](http://localhost:3000/?genome=hg38&gene=ENST00000407796&mds3=GDC&filterObj={%22type%22:%22tvslst%22,%22in%22:true,%22join%22:%22%22,%22lst%22:[{%22type%22:%22tvs%22,%22tvs%22:{%22term%22:{%22id%22:%22case.demographic.gender%22},%22values%22:[{%22key%22:%22female%22}],%22isnot%22:true}}]}) filtering, same reason as unable to use gene exp for sample filtering for gdc (api issue)

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [ ] Tests: added and/or passed unit and integration tests, or N/A
- [ ] Todos: commented or documented, or N/A
- [ ] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

File: server/src/mds3.gdc.js [hunk 1]
```diff
@@ -1282,15 +1284,110 @@ returns:
...
+			if (q.isHierCluster && !ds.__gdc.casesWithExpData.has(d.id)) {
+				// case does not have gene exp data; only cases with exp data are allowed
+				continue
+			}
+			s = { sample_id: d.id }
+			samples.push(s)
+		}
+		// assign survival data for this sample
+		s[survivalTwLst[0].term.id] = {
+			key: d.censored ? 1 : 0,
```
Comment (xzhou82) [inline server/src/mds3.gdc.js:1383]: @congyu-lu please see if the key value 1/0 is assigned correctly

Comment (congyu-lu) [reply server/src/mds3.gdc.js:1383]: I think censored generally has a code of 0 and event-occurred has a code of 1

**Comment by xzhou82** [comment]
please test now. oncomatrix and mds3 tests pass. it remains to be seen if survival data query works for any gdc cohort filter.

File: server/src/mds3.gdc.js [hunk 2]
```diff
@@ -1282,94 +1285,190 @@ returns:
...
+		const g = filter2GDCfilter(q.filterObj)
+		if (g) filter.content.push(g)
+	}
+	addTid2value_to_filter(q.tid2value, filter.content, ds)
+
+	// the survival term itself is not used in api query, since there's just one type of survival data from gdc and no need to distinguish
+	const { host, headers } = ds.getHostHeaders(q)
+	const re = await ky
+		// hardcoded url in production environment. TODO check if works at all for qa environments
+		.post('https://portal.gdc.cancer.gov/auth/api/v0/analysis/survival', {
```
Comment (xzhou82) [inline server/src/mds3.gdc.js:1357]: @siosonel please assess this

Comment (siosonel) [reply server/src/mds3.gdc.js:1357]: @xzhou82  The same URL should work in qa environments, it'll just point to different IP addresses depending on the VPN user

Comment (xzhou82) [reply server/src/mds3.gdc.js]: thanks replaced with `host.rest` and works

**Review by siosonel — APPROVED** [review]
Works, tested with http://localhost:3000/example.gdc.matrix.html?maxGenes=3&cohort=Gliomas, by adding Overall Survival term and editing to display as time-to-event.

**Pull request merged**